File: 冒泡排序/Codes/Bubble_Sort.py
nums = [1, 2, 8, 5, 3, 4, 7]


def bubblesort(nums: list):
    n = len(nums)
    b = True
    for i in range(n - 1):
        if b:
            b = False
            print(i)
            for j in range(n - i - 1):
                if nums[j] > nums[j + 1]:
                    nums[j], nums[j + 1] = nums[j + 1], nums[j]
                    b = True


bubblesort(nums)
print(nums)

File: 冒泡排序/Codes/Bubble_Sort.java
package com.Bubble_Sort;

import java.util.Arrays;

public class Bubble_Sort {
    public static void main(String[] args) {
        int[] nums = {1, 2, 8, 5, 3, 4, 7};
        bubbleSort(nums);
        System.out.println(Arrays.toString(nums));
    }

    private static void bubbleSort(int[] nums) {
        boolean b = true;
        for (int i = 0, len = nums.length; i < len - 1 && b; ++i) {
            b = false;
//            System.out.println(i);
            for (int j = 0; j < len - i - 1; j++) {
                if (nums[j] > nums[j + 1]) {
                    int n = nums[j];
                    nums[j] = nums[j + 1];
                    nums[j + 1] = n;
                    b = true;
                }
            }
        }
    }
}
